src/test/java/uk/gov/pay/directdebit/webhook/gocardless/services/GoCardlessPaymentHandlerTest.java

```diff
@@ -45,134 +43,132 @@
+        Assert.assertThat(storedGoCardlessEvent.getEventId(), is(event.getId()));
+        Assert.assertThat(storedGoCardlessEvent.getGoCardlessEventId(), is(goCardlessEvent.getGoCardlessEventId()));
     }
 
     @Test
     public void handle_onFailedPaymentGoCardlessEvent_shouldSetAPayEventAsFailed() {
         GoCardlessEvent goCardlessEvent = spy(GoCardlessEventFixture.aGoCardlessEventFixture().withAction("failed").toEntity());
 
         when(mockedGoCardlessService.findPaymentForEvent(goCardlessEvent)).thenReturn(goCardlessPaymentFixture.toEntity());
-        when(mockedTransactionService.paymentFailedWithEmailFor(transaction, payer)).thenReturn(paymentRequestEvent);
+        when(mockedTransactionService.paymentFailedWithEmailFor(transaction)).thenReturn(
+                event);
 
         goCardlessPaymentHandler.handle(goCardlessEvent);
 
-        verify(mockedTransactionService).paymentFailedWithEmailFor(transaction, payer);
-        verify(goCardlessEvent).setPaymentRequestEventId(paymentRequestEvent.getId());
-        verify(mockedGoCardlessService).storeEvent(geCaptor.capture());
+        verify(mockedTransactionService).paymentFailedWithEmailFor(transaction);
+        verify(goCardlessEvent).setEventId(event.getId());
+        verify(mockedGoCardlessService).updateInternalEventId(geCaptor.capture());
         GoCardlessEvent storedGoCardlessEvent = geCaptor.getValue();
-        Assert.assertThat(storedGoCardlessEvent.getPaymentRequestEventId(), is(paymentRequestEvent.getId()));
+        Assert.assertThat(storedGoCardlessEvent.getEventId(), is(event.getId()));
+        Assert.assertThat(storedGoCardlessEvent.getGoCardlessEventId(), is(goCardlessEvent.getGoCardlessEventId()));
     }
 
     @Test
     public void handle_onPayoutPaidGoCardlessEvent_shouldLinkTheEventToAnExistingPaymentPaidOutPayEvent() {
         GoCardlessEvent goCardlessEvent = spy(GoCardlessEventFixture.aGoCardlessEventFixture().withAction("paid").toEntity());
 
         when(mockedGoCardlessService.findPaymentForEvent(goCardlessEvent)).thenReturn(goCardlessPaymentFixture.toEntity());
-        when(mockedTransactionService.payoutPaidFor(transaction)).thenReturn(paymentRequestEvent);
+        when(mockedTransactionService.payoutPaidFor(transaction)).thenReturn(event);
 
         goCardlessPaymentHandler.handle(goCardlessEvent);
 
-        verify(goCardlessEvent).setPaymentRequestEventId(paymentRequestEvent.getId());
-        verify(mockedGoCardlessService).storeEvent(geCaptor.capture());
+        verify(goCardlessEvent).setEventId(event.getId());
+        verify(mockedGoCardlessService).updateInternalEventId(geCaptor.capture());
         GoCardlessEvent storedGoCardlessEvent = geCaptor.getValue();
-        Assert.assertThat(storedGoCardlessEvent.getPaymentRequestEventId(), is(paymentRequestEvent.getId()));
-    }
-
-    @Test
-    public void handle_onANotHandledPaymentGoCardlessEvent_shouldStoreAnUnlinkedGoCardlessEvent() {
```

Comment:
Again, not sure why this was removed

Reply:
because storing events at this level was unnecessary as it has been already stored in WebhookGoCardlessService (line 33). Too lazy to copy the link.